### PR TITLE
New /kernel/lifecycle page

### LINKED ISSUE #5684: [Kernel] Build ubuntu.com/kernel/lifecycle
Build the five pages that were sketched in [Ubuntu-design#120](https://github.com/CanonicalLtd/Ubuntu-design/issues/120) and [Ubuntu-design#121](https://github.com/CanonicalLtd/Ubuntu-design/issues/121):

URL|Title + copydoc
-|-
ubuntu.com/kernel/lifecycle|[Ubuntu kernel lifecycle and enablement stack](https://docs.google.com/document/d/1d01SFs7c56ge2o92g6Rwr3SjG8rtxm3jNwIyL1N8mQg/edit)

I’ve proposed fixes for the worst, but not all, of the language issues in the source documents. Ideally each copydoc would be reviewed by someone doing pair-editing with a kernel engineer for accuracy.

This issue might need converting to an epic for building each page separately. If so, take care that the “Kernel” navigation ribbon is coherent at each step.

### PR DESCRIPTION
**Pull request by pmahnke**
## Done

- Created a new /kernel/lifecycle page

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: http://0.0.0.0:8001/kernel/lifecycle
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Compare the text to the [copy doc](https://docs.google.com/document/d/1d01SFs7c56ge2o92g6Rwr3SjG8rtxm3jNwIyL1N8mQg/edit#heading=h.3gbb9wcbk9us) and the charts to the [sheet](https://docs.google.com/spreadsheets/d/1KIuOAa620yseE0uhIl67u2SlS0bWlPI_d98bvYM2Yl8/edit#gid=703230516)

## TODO

- [x] Possibly put the charts in a menu like the instructions
- [x] Possibly reverse the order of versions to match the instructions
- [ ] Get the data for 18.04 and in general confirmed by Lech and Brad

## Issue / Card

Fixes #5684

### CODE REVIEW

File: templates/kernel/lifecycle.html [hunk 1]
```diff
@@ -0,0 +1,812 @@
...
+        Installation
+      </h2>
+
+      Show instructions for:
+      <a tabindex="1" class="p-menu-18-04">18.04 LTS</a> -
+      <a tabindex="2" class="p-menu-16-04">16.04 LTS</a> -
+      <a tabindex="3" class="p-menu-14-04">14.04 LTS</a> -
+      <a tabindex="4" class="p-menu-12-04">12.04 LTS</a>
+
+      <div class="" id="inst-18-04">
```
Comment (anthonydillon) [inline templates/kernel/lifecycle.html]: Remove empty class please

File: templates/kernel/lifecycle.html [hunk 2]
```diff
@@ -0,0 +1,812 @@
...
+</section>
+
+<section class="p-strip--light is-bordered">
+  <div class="row">
+    <div class="col-8">
+      <h2>
+        Installation
+      </h2>
+
+      Show instructions for:
```
Comment (anthonydillon) [inline templates/kernel/lifecycle.html]: Should this be in a `p`?

File: templates/kernel/lifecycle.html [hunk 3]
```diff
@@ -0,0 +1,812 @@
...
+        </p>
+        <pre><code>sudo apt-get install --install-recommends linux-generic-hwe-18.04</code></pre>
+        <p>
+          The 18.04.2 and newer point releases will ship with an updated kernel and X stack by default for the desktop. Server installations will default to the GA kernel and provide the enablement kernel as optional.</p>
+          <p>
+            The 18.04 HWE Stacks will follow the Rolling Update Model.
+          </p>
+        </div>
+
+        <div class="" id="inst-16-04">
```
Comment (anthonydillon) [inline templates/kernel/lifecycle.html]: Remove empty class

File: templates/kernel/lifecycle.html [hunk 4]
```diff
@@ -0,0 +1,812 @@
...
+          <p>
+            Server:
+          </p>
+          <pre><code>sudo apt-get install --install-recommends linux-generic-hwe-16.04</code></pre>
+          <p>
+            The 16.04.2 and newer point releases ship with an updated kernel and X stack by default for the desktop. Server installations default to the GA kernel and provide the enablement kernel as optional.The 16.04 HWE Stacks follow the Rolling Update Model.
+          </p>
+        </div>
+
+        <div class="" id="inst-14-04">
```
Comment (anthonydillon) [inline templates/kernel/lifecycle.html]: Remove empty class

File: templates/kernel/lifecycle.html [hunk 5]
```diff
@@ -0,0 +1,812 @@
...
+          <p>
+            Server:
+          </p>
+          <pre><code>sudo apt-get install --install-recommends linux-generic-lts-xenial</code></pre>
+          <p>
+            The 14.04.2 and newer point releases ship with an updated kernel and X stack by default. If you have installed with older media, you can use these instructions to install the newer HWE kernel derived from 16.04 (Xenial).
+          </p>
+        </div>
+
+        <div class="" id="inst-12-04">
```
Comment (anthonydillon) [inline templates/kernel/lifecycle.html]: Remove empty class

File: templates/kernel/lifecycle.html [hunk 6]
```diff
@@ -0,0 +1,812 @@
...
+          Show schedule for:
+        </p>
+        <a tabindex="5" class="p-menu-all">All releases</a> -
+        <a tabindex="6" class="p-menu-lts">All LTS releases</a> -
+        <a tabindex="7" class="p-menu-20-04">20.04 LTS</a> -
+        <a tabindex="8" class="p-menu-18-04">18.04 LTS</a> -
+        <a tabindex="9" class="p-menu-16-04">16.04 LTS</a> -
+        <a tabindex="10" class="p-menu-14-04">14.04 LTS</a>
+
+        <div class="" id="support-20-04">
```
Comment (anthonydillon) [inline templates/kernel/lifecycle.html]: Remove empty class

File: templates/kernel/lifecycle.html [hunk 7]
```diff
@@ -0,0 +1,812 @@
...
+                <td><strong>Ubuntu 20.04.0 LTS</strong></td>
+                <td>Apr 2020</td>
+                <td>Apr 2025</td>
+                <td>Apr 2030</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+
+        <div class="" id="support-18-04">
```
Comment (anthonydillon) [inline templates/kernel/lifecycle.html]: Remove empty class

File: templates/kernel/lifecycle.html [hunk 8]
```diff
@@ -0,0 +1,812 @@
...
+                <td><strong>Ubuntu 18.04.0 LTS (v4.15)</strong></td>
+                <td>Apr 2018</td>
+                <td>Apr 2023</td>
+                <td>Apr 2028</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+
+        <div class="" id="support-16-04">
```
Comment (anthonydillon) [inline templates/kernel/lifecycle.html]: Remove empty class

File: templates/kernel/lifecycle.html [hunk 9]
```diff
@@ -0,0 +1,812 @@
...
+                <td><strong>Ubuntu 16.04.0 LTS (v4.4)</strong></td>
+                <td>Apr 2016</td>
+                <td>Apr 2021</td>
+                <td>Apr 2024</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+
+        <div class="" id="support-14-04">
```
Comment (anthonydillon) [inline templates/kernel/lifecycle.html]: Remove empty class

File: templates/kernel/lifecycle.html [hunk 10]
```diff
@@ -0,0 +1,812 @@
...
+                <td><strong>Ubuntu 14.04.0 LTS (v3.13)</strong></td>
+                <td>Apr 2014</td>
+                <td>Apr 2019</td>
+                <td>Apr 2022</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+
+        <div class="" id="support-all">
```
Comment (anthonydillon) [inline templates/kernel/lifecycle.html]: Remove empty class

File: templates/kernel/lifecycle.html [hunk 11]
```diff
@@ -0,0 +1,812 @@
...
+                <td><strong>Ubuntu 20.04.5 LTS</strong></td>
+                <td>May 2022</td>
+                <td>Aug 2022</td>
+                <td>Apr 2025</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+
+        <div class="" id="support-lts">
```
Comment (anthonydillon) [inline templates/kernel/lifecycle.html]: Remove empty class

File: static/sass/_pattern_kernel_lifecycle.scss [hunk 12]
```diff
@@ -0,0 +1,36 @@
+// css menu for the /kernel/lifecycle page
+
+@mixin ubuntu-p-kernel-lifecycle {
+
+  .p-insta--hide {
```
Comment (anthonydillon) [inline static/sass/_pattern_kernel_lifecycle.scss]: Can't you just use `u-hide`?

File: templates/kernel/lifecycle.html [hunk 13]
```diff
@@ -0,0 +1,812 @@
...
+
+<section class="p-strip--light is-bordered">
+  <div class="row">
+    <div class="col-8">
+      <h2>
+        Installation
+      </h2>
+
+      Show instructions for:
+      <a tabindex="1" class="p-menu-18-04">18.04 LTS</a> -
```
Comment (anthonydillon) [inline templates/kernel/lifecycle.html]: Can I suggest you use the tab component? It's unclear what is going to happen when selecting an option. https://docs.vanillaframework.io/patterns/tabs

**Comment by anthonydillon** [comment]
@pmahnke fixed up the charts. Tricky one. Also, added some extra aria stuff. 

Could you double-check the content of the panels before merging? Some seem wrong,

**Comment by pmahnke** [comment]
@anthonydillon seems the instructions are correct.  Can you rereview?

**Comment by anthonydillon** [comment]
QA issue the inital selected installation panel and tab does not match up.
image

File: templates/kernel/lifecycle.html [hunk 14]
```diff
@@ -0,0 +1,784 @@
...
+
+      <div class="p-tabs__content" id="installation-12-04" role="tabpanel" aria-labelledby="installation-12-04-tab">
+        <h2>Ubuntu 12.04 LTS &mdash; Precise Pangolin</h2>
+        <p>Desktop:</p>
+        <pre><code>sudo apt-get install --install-recommends linux-generic-lts-trusty xserver-xorg-lts-trusty libgl1-mesa-glx-lts-trusty</code></pre>
+        <p>Server:</p>
+        <pre><code>sudo apt-get install --install-recommends linux-generic-lts-trusty</code></pre>
+        <p>The 12.04.2 and newer point releases shipped with an updated kernel and X stack by default. The hardware enablement stack is comprised of the newer kernel and X stacks from 14.04 (Trusty).</p>
+      </div>
+    </div>
```
Comment (anthonydillon) [inline templates/kernel/lifecycle.html]: Missing a closing div here

File: templates/kernel/lifecycle.html [hunk 15]
```diff
@@ -0,0 +1,784 @@
...
+              <th scope="col">Released</th>
+              <th scope="col">End of Life</th>
+              <th scope="col">Extended security maintenance</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td><strong>Ubuntu 14.04 LTS (v3.13)</strong></td>
+              <td>Apr 2014</td>
+              <td>Apr 2019</td>
```
Comment (anthonydillon) [inline templates/kernel/lifecycle.html]: Should match the columns in the head so add `<td>&nbsp;</td>` to the end here for each row.

File: templates/kernel/lifecycle.html [hunk 16]
```diff
@@ -0,0 +1,784 @@
...
+            <li class="p-tabs__item" role="presentation">
+              <a href="#support-16-04" class="p-tabs__link" role="tab" id="support-16-04-tab" aria-controls="support-16-04">16.04 LTS</a>
+            </li>
+            <li class="p-tabs__item" role="presentation">
+              <a href="#support-14-04" class="p-tabs__link" role="tab" id="support-14-04-tab" aria-controls="support-14-04">14.04 LTS</a>
+            </li>
+          </ul>
+        </nav>
+
+        <div class="p-tabs__content" id="support-20-04" role="tabpanel" aria-labelledby="installation-20-04-tab">
```
Comment (anthonydillon) [inline templates/kernel/lifecycle.html]: This should be `aria-labelledby="support-20-04-tab"` and the same for the others.

**Comment by anthonydillon** [comment]
The QA issue is still present.

File: templates/kernel/lifecycle.html [hunk 17]
```diff
@@ -0,0 +1,784 @@
...
+      </div>
+    </div>
+  </div>
+</section>
+
+<section class="p-strip is-bordered">
+  <div class="row">
+    <div class="col-8">
+      <h2>Check your support status</h2>
+      <p>To determine if your install is still supported, use <code>hwe-support-status</code> as documented on the <a href="/about/release-cycle">Ubuntu release cycle</a>.</p>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]:
```suggestion
      <p>To determine if your install is still supported, use <code>hwe-support-status</code>:</p>
```

It’s not documented on that page, nor even proposed for it. (And even if it was, we shouldn’t be sending people to a separate page for a single terminal command.)

File: templates/kernel/lifecycle.html [hunk 18]
```diff
@@ -0,0 +1,784 @@
...
+              <td>Apr 2025</td>
+            </tr>
+          </tbody>
+        </table>
+      </div>
+
+      <div class="p-tabs__content" id="support-all" role="tabpanel" aria-labelledby="support-all-tab">
+        <h3>Ubuntu Kernel Support</h3>
+        <div class="u-hide--small" id="kernelall"></div>
+        <p>This is a combined view of the diagrams from above (ie the Ubuntu Kernel Release Schedule combined with the Distilled 14.04.x, 16.04.x, and 18.04.x Ubuntu Kernel Support diagrams)</p>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]: I’m sorry I didn’t pick this up when editing the copydoc earlier: Whenever you can see this text, the diagram is a combined view of some diagrams, sure — but none of them are present on the page, so they’re not “above”.

I think the message this paragraph is trying to convey is already conveyed, more concisely, by the tab selector. So the paragraph can be dropped altogether.
```suggestion
```

**Comment by pmahnke** [comment]
@anthonydillon the tabs are a bit funny because there are two sets on the page.  Any change and it will unset the other set of tabs focus.

Any thoughts?

**Comment by anthonydillon** [comment]
Yes, you should treat each tab set individually. Therefore when clicking a tab link you should use closest to find the container p-tab then find all the links in there and only effect them.

**Comment by anthonydillon** [comment]
@pmahnke fixed up the tabs stuff

**Comment by anasereijo** [comment]
@pmahnke looks good to me! just 1 tiny thing:
Screenshot 2020-02-24 at 09 57 59

File: templates/kernel/lifecycle.html [hunk 19]
```diff
@@ -0,0 +1,747 @@
...
+              <a href="#installation-14-04" class="p-tabs__link" id="installation-14-04-tab" role="tab" aria-controls="installation-14-04">14.04 LTS</a>
+            </li>
+            <li class="p-tabs__item" role="presentation">
+              <a href="#installation-12-04" class="p-tabs__link" id="installation-12-04-tab" role="tab" aria-controls="installation-12-04">12.04 LTS</a>
+            </li>
+          </ul>
+        </nav>
+
+        <div class="p-tabs__content" id="installation-12-04" role="tabpanel" aria-labelledby="installation-12-04-tab">
+          <h2>Ubuntu 12.04 LTS &mdash; Precise Pangolin</h2>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]:
```suggestion
          <p>Ubuntu 12.04 LTS &mdash; Precise Pangolin</p>
```

File: templates/kernel/lifecycle.html [hunk 20]
```diff
@@ -0,0 +1,747 @@
...
+          <p>Desktop:</p>
+          <pre><code>sudo apt-get install --install-recommends linux-generic-lts-trusty xserver-xorg-lts-trusty libgl1-mesa-glx-lts-trusty</code></pre>
+          <p>Server:</p>
+          <pre><code>sudo apt-get install --install-recommends linux-generic-lts-trusty</code></pre>
+          <p>The 12.04.2 and newer point releases shipped with an updated kernel and X stack by default. The hardware enablement stack is comprised of the newer kernel and X stacks from 14.04 (Trusty).</p>
+        </div>
+
+        <div class="p-tabs__content" id="installation-14-04" role="tabpanel" aria-labelledby="installation-14-04-tab">
+          <h2>Ubuntu 14.04 LTS &mdash; Trusty Tahr</h2>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]:
```suggestion
          <p>Ubuntu 14.04 LTS &mdash; Trusty Tahr</p>
```

File: templates/kernel/lifecycle.html [hunk 21]
```diff
@@ -0,0 +1,747 @@
...
+          <p>Server:</p>
+          <pre><code>sudo apt-get install --install-recommends linux-generic-lts-xenial</code></pre>
+          <p>The 14.04.2 and newer point releases ship with an updated kernel and X stack by default. If you have installed with older media, you can use these instructions to install the newer HWE kernel derived from 16.04 (Xenial).</p>
+          <p>If you run a multiarch desktop (for example, i386 and amd64 on amd64, for gaming or Wine), you may find you need a slightly more involved command:</p>
+          <pre><code>sudo apt-get install --install-recommends linux-generic-lts-xenial xserver-xorg-core-lts-xenial xserver-xorg-lts-xenial xserver-xorg-video-all-lts-xenial xserver-xorg-input-all-lts-xenial libwayland-egl1-mesa-lts-xenial libgl1-mesa-glx-lts-xenial libgl1-mesa-glx-lts-xenial:i386 libglapi-mesa-lts-xenial:i386</code></pre>
+        </div>
+
+        <div class="p-tabs__content" id="installation-16-04" role="tabpanel" aria-labelledby="installation-16-04-tab">
+          <h2>Ubuntu 16.04 LTS &mdash; Xenial Xerus</h2>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]:
```suggestion
          <p>Ubuntu 16.04 LTS &mdash; Xenial Xerus</p>
```

File: templates/kernel/lifecycle.html [hunk 22]
```diff
@@ -0,0 +1,747 @@
...
+          <p>Desktop:</p>
+          <pre><code>sudo apt-get install --install-recommends linux-generic-hwe-16.04 xserver-xorg-hwe-16.04</code></pre>
+          <p>Server:</p>
+          <pre><code>sudo apt-get install --install-recommends linux-generic-hwe-16.04</code></pre>
+          <p>The 16.04.2 and newer point releases ship with an updated kernel and X stack by default for the desktop. Server installations default to the GA kernel and provide the enablement kernel as optional.The 16.04 HWE Stacks follow the Rolling Update Model.</p>
+        </div>
+
+        <div class="p-tabs__content" id="installation-18-04" role="tabpanel" aria-labelledby="installation-18-04-tab">
+          <h2>Ubuntu 18.04 LTS &mdash; Bionic Beaver</h2>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]:
```suggestion
          <p>Ubuntu 18.04 LTS &mdash; Bionic Beaver</p>
```

File: templates/kernel/lifecycle.html [hunk 23]
```diff
@@ -0,0 +1,747 @@
...
+            <li class="p-tabs__item" role="presentation">
+              <a href="#support-14-04" class="p-tabs__link" role="tab" id="support-14-04-tab" aria-controls="support-14-04">14.04 LTS</a>
+            </li>
+          </ul>
+        </nav>
+
+        <div class="p-tabs__content" id="support-20-04" role="tabpanel" aria-labelledby="support-20-04-tab">
+          <h3>20.04.x Ubuntu kernel support</h3>
+          <div class="u-hide--small" id="kernel2004"></div>
+          <p>This is a distilled view of our 20.04.x Ubuntu kernel support schedule.</p>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]:
```suggestion
```

File: templates/kernel/lifecycle.html [hunk 24]
```diff
@@ -0,0 +1,747 @@
...
+                <td>Apr 2030</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+
+        <div class="p-tabs__content" id="support-18-04" role="tabpanel" aria-labelledby="support-18-04-tab">
+          <h3>18.04.x Ubuntu kernel support</h3>
+          <div class="u-hide--small" id="kernel1804"></div>
+          <p>This is a distilled view of our 18.04.x Ubuntu kernel support schedule.</p>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]:
```suggestion
```

File: templates/kernel/lifecycle.html [hunk 25]
```diff
@@ -0,0 +1,747 @@
...
+                <td>Apr 2028</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+
+        <div class="p-tabs__content" id="support-16-04" role="tabpanel" aria-labelledby="support-16-04-tab">
+          <h3>16.04.x Ubuntu kernel support</h3>
+          <div class="u-hide--small" id="kernel1604"></div>
+          <p>This is a distilled view of our 16.04.x Ubuntu kernel support schedule.</p>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]:
```suggestion
```

File: templates/kernel/lifecycle.html [hunk 26]
```diff
@@ -0,0 +1,747 @@
...
+                <td>Apr 2024</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+
+        <div class="p-tabs__content" id="support-14-04" role="tabpanel" aria-labelledby="support-14-04-tab">
+          <h3>14.04.x Ubuntu kernel support</h3>
+          <div class="u-hide--small" id="kernel1404"></div>
+          <p>This is a distilled view of our 14.04.x Ubuntu kernel support schedule.</p>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]:
```suggestion
```

File: templates/kernel/lifecycle.html [hunk 27]
```diff
@@ -0,0 +1,747 @@
...
+                <td>Apr 2022</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+
+        <div class="p-tabs__content" id="support-lts" role="tabpanel" aria-labelledby="support-lts-tab">
+          <h3>All LTS releases</h3>
+          <div class="u-hide--small" id="kernellts"></div>
+          <p>A more detailed view of our LTS kernel support schedule. This includes views for 14.04.x, 16.04.x and 18.04.x as well as upstream stable commitments.</p>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]:
```suggestion
```

File: templates/kernel/lifecycle.html [hunk 28]
```diff
@@ -0,0 +1,747 @@
...
+                <td>Apr 2014</td>
+                <td>Apr 2019</td>
+                <td>Apr 2022</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+
+        <div class="p-tabs__content" id="support-lts" role="tabpanel" aria-labelledby="support-lts-tab">
+          <h3>All LTS releases</h3>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]:
```suggestion
```

File: templates/kernel/lifecycle.html [hunk 29]
```diff
@@ -0,0 +1,747 @@
...
+        </tbody>
+      </table>
+    </div>
+  </div>
+</section>
+
+<section class="p-strip is-bordered">
+  <div class="row">
+    <div class="col-8">
+      <h3>Kernel support schedule</h3>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]: As pointed out by Ana:
```suggestion
      <h2>Kernel support schedule</h2>
```

File: templates/kernel/lifecycle.html [hunk 30]
```diff
@@ -0,0 +1,747 @@
...
+              <a href="#support-16-04" class="p-tabs__link" role="tab" id="support-16-04-tab" aria-controls="support-16-04">16.04 LTS</a>
+            </li>
+            <li class="p-tabs__item" role="presentation">
+              <a href="#support-14-04" class="p-tabs__link" role="tab" id="support-14-04-tab" aria-controls="support-14-04">14.04 LTS</a>
+            </li>
+          </ul>
+        </nav>
+
+        <div class="p-tabs__content" id="support-20-04" role="tabpanel" aria-labelledby="support-20-04-tab">
+          <h3>20.04.x Ubuntu kernel support</h3>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]:
```suggestion
```

File: templates/kernel/lifecycle.html [hunk 31]
```diff
@@ -0,0 +1,747 @@
...
+                <td>Apr 2020</td>
+                <td>Apr 2025</td>
+                <td>Apr 2030</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+
+        <div class="p-tabs__content" id="support-18-04" role="tabpanel" aria-labelledby="support-18-04-tab">
+          <h3>18.04.x Ubuntu kernel support</h3>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]:
```suggestion
```

File: templates/kernel/lifecycle.html [hunk 32]
```diff
@@ -0,0 +1,747 @@
...
+                <td>Apr 2018</td>
+                <td>Apr 2023</td>
+                <td>Apr 2028</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+
+        <div class="p-tabs__content" id="support-16-04" role="tabpanel" aria-labelledby="support-16-04-tab">
+          <h3>16.04.x Ubuntu kernel support</h3>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]:
```suggestion
```

File: templates/kernel/lifecycle.html [hunk 33]
```diff
@@ -0,0 +1,747 @@
...
+                <td>Apr 2016</td>
+                <td>Apr 2021</td>
+                <td>Apr 2024</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+
+        <div class="p-tabs__content" id="support-14-04" role="tabpanel" aria-labelledby="support-14-04-tab">
+          <h3>14.04.x Ubuntu kernel support</h3>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]:
```suggestion
```

File: templates/kernel/lifecycle.html [hunk 34]
```diff
@@ -0,0 +1,747 @@
...
+      <h3>Kernel support schedule</h3>
+
+      <div class="js-tab-container">
+        <nav class="p-tabs">
+          <ul class="p-tabs__list" role="tablist">
+            <li class="p-tabs__item" role="presentation">
+              <a href="#support-all" class="p-tabs__link" role="tab" id="support-all-tab" aria-controls="support-all" aria-selected="true">All releases</a>
+            </li>
+            <li class="p-tabs__item" role="presentation">
+              <a href="#support-lts" class="p-tabs__link" role="tab" id="support-lts-tab" aria-controls="support-lts">All LTS release</a>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]:
```suggestion
              <a href="#support-lts" class="p-tabs__link" role="tab" id="support-lts-tab" aria-controls="support-lts">All LTS releases</a>
```

File: templates/kernel/lifecycle.html [hunk 35]
```diff
@@ -0,0 +1,747 @@
...
+                <td>Aug 2022</td>
+                <td>Aug 2024</td>
+                <td>Apr 2025</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+
+        <div class="p-tabs__content" id="support-all" role="tabpanel" aria-labelledby="support-all-tab">
+          <h3>Ubuntu kernel support</h3>
```
Comment (matthewpaulthomas) [inline templates/kernel/lifecycle.html]:
```suggestion
```

**Pull request merged**